### ISSUE TITLE
[Bug] Can't launch any game with Steam Proton

### ISSUE THREAD

**Issue by crusardri** [redromnon/HeroicBashLauncher#60]
### Before posting this issue, make sure you've read the [Issues section in the Readme](https://github.com/redromnon/HeroicBashLauncher#issues-and-suggestions) and the [Flatpak/Steam-Deck guide](https://github.com/redromnon/HeroicBashLauncher/wiki/Steam-Deck-(Flatpak)-Guide).

**Describe the bug**
I can launch games from Heroic Game Launcher, but i can't launch any game inyected by HeroicBashLauncher 

**To Reproduce**
1. Execute setup.sh
2. Execute any game.sh
3. Error

1. Execute setup.sh
2. Launch Steam Deck Mode
3. Launch any inyected game
4. Error

**Expected behavior**
The game is launched

**Screenshots**

**Details**

 - Platform: Steam Deck
 - OS: SteamOS Holo 3.2
 - Heroic Version: v2.3.0-beta.3 Flatpak
 - Bash Launcher Version: v2.5.0 Flatpak
 - Proton 7.0

**Additional context**
Log file:
`
Gtk-Message: 13:34:54.553: Failed to load module "canberra-gtk-module"
3.32.0

Running launch command for Absolute Drift:
STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/deck/.var/app/com.heroicgameslauncher.hgl/.steam/steam STEAM_COMPAT_DATA_PATH=/AbsoluteDrift /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch 19927295d6e3467887d4e830d8c85963 --no-wine --wrapper "/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton run" 
[cli] INFO: Logging in...
[Core] INFO: Trying to re-use existing login session...
[cli] INFO: Checking for updates...
[Core] INFO: Getting authentication token...
[cli] INFO: Launching 19927295d6e3467887d4e830d8c85963...
Traceback (most recent call last):
  File "legendary/cli.py", line 3004, in <module>
  File "legendary/cli.py", line 2919, in main
  File "legendary/cli.py", line 685, in launch_game
  File "subprocess.py", line 858, in __init__
  File "subprocess.py", line 1704, in _execute_child
FileNotFoundError: [Errno 2] No such file or directory: '/home/deck/.steam/steam/steamapps/common/Proton'
[9] Failed to execute script 'cli' due to unhandled exception!
---CANNOT CONNECT TO NETWORK. RUNNING IN OFFLINE MODE---
[cli] INFO: Launching 19927295d6e3467887d4e830d8c85963...
Traceback (most recent call last):
  File "legendary/cli.py", line 3004, in <module>
  File "legendary/cli.py", line 2919, in main
  File "legendary/cli.py", line 685, in launch_game
  File "subprocess.py", line 858, in __init__
  File "subprocess.py", line 1704, in _execute_child
FileNotFoundError: [Errno 2] No such file or directory: '/home/deck/.steam/steam/steamapps/common/Proton'
[15] Failed to execute script 'cli' due to unhandled exception!
Gtk-Message: 13:34:56.694: Failed to load module "canberra-gtk-module"

** (zenity:19): WARNING **: 13:34:56.817: AT-SPI: Could not obtain desktop path or name

** (zenity:19): WARNING **: 13:34:56.831: AT-SPI: Could not obtain desktop path or name

** (zenity:19): WARNING **: 13:34:56.831: atk-bridge: get_device_events_reply: unknown signature

** (zenity:19): WARNING **: 13:34:56.832: atk-bridge: get_device_events_reply: unknown signature

** (zenity:19): WARNING **: 13:34:56.832: atk-bridge: GetRegisteredEvents returned message with unknown signature

** (zenity:19): WARNING **: 13:34:56.843: AT-SPI: Could not obtain desktop path or name

** (zenity:19): WARNING **: 13:34:56.844: atk-bridge: get_device_events_reply: unknown signature

** (zenity:19): WARNING **: 13:34:56.844: atk-bridge: get_device_events_reply: unknown signature

** (zenity:19): WARNING **: 13:34:56.844: atk-bridge: GetRegisteredEvents returned message with unknown signature
Using Bash Launcher 2.5.0

Checking if Zenity is installed:
`

`
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
3.32.0
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.

Running launch command for Death Stranding:
STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/deck/.var/app/com.heroicgameslauncher.hgl/.steam/steam STEAM_COMPAT_DATA_PATH=/DeathStranding /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch Boga --no-wine --wrapper "/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton run" 
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
[Core] WARNING: Could not determine locale, falling back to en-US
[cli] INFO: Logging in...
[Core] INFO: Trying to re-use existing login session...
[cli] INFO: Checking for updates...
[Core] INFO: Getting authentication token...
[Core] INFO: Getting ownership token.
[cli] INFO: Launching Boga...
Traceback (most recent call last):
  File "legendary/cli.py", line 3004, in <module>
  File "legendary/cli.py", line 2919, in main
  File "legendary/cli.py", line 685, in launch_game
  File "subprocess.py", line 858, in __init__
  File "subprocess.py", line 1704, in _execute_child
FileNotFoundError: [Errno 2] No such file or directory: '/home/deck/.steam/steam/steamapps/common/Proton'
[21] Failed to execute script 'cli' due to unhandled exception!
---CANNOT CONNECT TO NETWORK. RUNNING IN OFFLINE MODE---
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
[Core] WARNING: Could not determine locale, falling back to en-US
[cli] INFO: Launching Boga...
Traceback (most recent call last):
  File "legendary/cli.py", line 3004, in <module>
  File "legendary/cli.py", line 2919, in main
  File "legendary/cli.py", line 685, in launch_game
  File "subprocess.py", line 858, in __init__
  File "subprocess.py", line 1704, in _execute_child
FileNotFoundError: [Errno 2] No such file or directory: '/home/deck/.steam/steam/steamapps/common/Proton'
[27] Failed to execute script 'cli' due to unhandled exception!
pid 25 != 18, skipping destruction (fork without exec?)
pid 19 != 18, skipping destruction (fork without exec?)
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
Using Bash Launcher 2.5.0

Checking if Zenity is installed:
`

**Comment by redromnon** [comment]
The thing is that in Linux, the terminal reads the file path as two separate things if it finds a space in it.

You have two options :
- For now the easier way, i.e. without editing json files for each games, is using **Proton-GE** instead by downloading it from Heroic's Wine Manager. 
- Or go to `.var/app/com.heroicgameslauncher.hgl/config/heroic/GamesConfig` -> look for "bin" -> add a backslash like `"bin": "/home/deck/.steam/steam/steamapps/common/Proton\\ 7.0/proton"` to your game's .json file.

**Comment by clemthi** [comment]
Surprisingly, same games (e.g. Shadow Tactics) having a space in the `bin` path do work but other (e.g. Death's Door) don't.

**Comment by mjholtzem** [comment]
I've tried using Proton-GE and gotten the same error. Also tried adding the backslash to the games json file and I get the following error

`FileNotFoundError: [Errno 2] No such file or directory: '/home/deck/.steam/steam/steamapps/common/Proton\\ 7.0/proton'`

**Comment by mjholtzem** [comment]
> > I've tried using Proton-GE and gotten the same error. Also tried adding the backslash to the games json file and I get the following error
> > `FileNotFoundError: [Errno 2] No such file or directory: '/home/deck/.steam/steam/steamapps/common/Proton\\ 7.0/proton'`
> 
> Try to create a symlink of proton 7.0 without spaces, and change it on heroic launcher config

created the symlink and it showed up as an option in Heroic. Assigned it and checked that the config got updated with the new path. Ran the launch script again and still got the same FileNotFoudn Error

`FileNotFoundError: [Errno 2] No such file or directory: '/home/deck/.steam/steam/steamapps/common/Proton7.0/proton`

I also tried downloading and selecting Wine-GE-Proton7-11 which I actually had some success with. One game was all pink when it launched but I succesfully launched a couple others and did not get the FileNotFound error

**Comment by redromnon** [comment]
> I've tried using Proton-GE and gotten the same error. Also tried adding the backslash to the games json file and I get the following error
> 
> `FileNotFoundError: [Errno 2] No such file or directory: '/home/deck/.steam/steam/steamapps/common/Proton\\ 7.0/proton'`

But Proton-GE doesn't have any spaces in between.

**Comment by redromnon** [comment]
Try using single quotes around Proton 7 and check by editing the game's launch script(.sh) file
Like `--no-wine --wrapper "/home/deck/.steam/steam/steamapps/common/'Proton 7.0'/proton run"`

**Comment by carn1x** [comment]
> Try using single quotes around Proton 7 and check by editing the game's launch script(.sh) file Like `--no-wine --wrapper "/home/deck/.steam/steam/steamapps/common/'Proton 7.0'/proton run"` or even the entire path like `--no-wine --wrapper "'/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton' run"`

I'm not sure what you mean by this, but I instead tried to wrap `/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton` as mentioned in the GamesConfig json file with single quotes inside the double quotes. This advances the error message so that this message no longer occurs:

```
 FileNotFoundError: [Errno 2] No such file or directory: '/home/deck/.steam/steam/steamapps/common/Proton'
 ```
 
 And instead I'm now getting:
 
 ```
 FileNotFoundError: [Errno 2] No such file or directory: '/Yokus-Island-Express/pfx.lock'
 ```

I'm getting it for other games. I also get this new error when using Proton-GE without needing to manually adjust the GameConfig json file I assume since Proton-GE has no space in the path in the first place.

All games affected run fine through Heroic Launcher still.

**Comment by redromnon** [comment]
Can you check the game_Heroic.log in `HeroicBashLauncher/GameFiles/log` folder for the launch command? Has the change been reflected there? You should see the Proton path in quotes.

**Comment by carn1x** [comment]
> Are you using HBL v2.5.0? I assumed you were using the beta version where the launch command resides in the launch scripts generated by Bash Launcher itself.
> 
> But I find it odd. It should've worked.

I'm using 2.5.0 Flatpak version, and its the only version I've used n case there's a chance of old version debris. I don't see any mention of proton in the `setup.sh` generated launcher scripts so I took a look in GameConfig instead on a whim since it was mentioned in the generated launcher scripts.

The command is not listed in the log file, here it is in full:

```
Gtk-Message: 14:09:38.777: Failed to load module "canberra-gtk-module"
3.32.0
 
 
Running launch command for Yokus[cli] INFO: Logging in...
[Core] INFO: Trying to re-use existing login session...
[cli] INFO: Checking for updates...
[Core] INFO: Getting authentication token...
[cli] INFO: Launching 3acb0920074446e8baaa4c249200d892...
sh: -c: line 5: unexpected EOF while looking for matching `''
sh: -c: line 6: syntax error: unexpected end of file
Traceback (most recent call last):
  File "/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton", line 1392, in <module>
    g_session.init_session(sys.argv[1] != "runinprefix")
  File "/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton", line 1194, in init_session
    g_compatdata.setup_prefix()
  File "/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton", line 704, in setup_prefix
    with self.prefix_lock:
  File "/home/deck/.local/share/Steam/steamapps/common/Proton 7.0/filelock.py", line 323, in __enter__
    self.acquire()
  File "/home/deck/.local/share/Steam/steamapps/common/Proton 7.0/filelock.py", line 271, in acquire
    self._acquire()
  File "/home/deck/.local/share/Steam/steamapps/common/Proton 7.0/filelock.py", line 384, in _acquire
    fd = os.open(self._lock_file, open_mode)
FileNotFoundError: [Errno 2] No such file or directory: '/Yokus-Island-Express/pfx.lock'
[cli] INFO: Logging in...
[Core] INFO: Trying to re-use existing login session...
[cli] INFO: Checking for updates...
[Core] INFO: Getting authentication token...
[cli] INFO: Launching 3acb0920074446e8baaa4c249200d892...
Using Bash Launcher 2.5.0
 
Checking if Zenity is installed:
Traceback (most recent call last):
  File "/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton", line 1392, in <module>
    g_session.init_session(sys.argv[1] != "runinprefix")
  File "/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton", line 1194, in init_session
    g_compatdata.setup_prefix()
  File "/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton", line 704, in setup_prefix
    with self.prefix_lock:
  File "/home/deck/.local/share/Steam/steamapps/common/Proton 7.0/filelock.py", line 323, in __enter__
    self.acquire()
  File "/home/deck/.local/share/Steam/steamapps/common/Proton 7.0/filelock.py", line 271, in acquire
    self._acquire()
  File "/home/deck/.local/share/Steam/steamapps/common/Proton 7.0/filelock.py", line 384, in _acquire
    fd = os.open(self._lock_file, open_mode)
FileNotFoundError: [Errno 2] No such file or directory: '/Yokus-Island-Express/pfx.lock'
```

**Comment by redromnon** [comment]
This is getting even stranger. I just tried HBL Flatpak v2.5.0 and it does mention the launch command in the log file for me.

```
Gtk-Message: 19:14:27.593: Failed to load module "xapp-gtk3-module"
Gtk-Message: 19:14:27.593: Failed to load module "xapp-gtk3-module"
Gtk-Message: 19:14:27.593: Failed to load module "canberra-gtk-module"
Gtk-Message: 19:14:27.594: Failed to load module "canberra-gtk-module"
3.32.0


Running launch command for Kingdom New Lands:
PULSE_LATENCY_MSEC=60 /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch cacao --wine /home/redromnon/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools/wine/Wine-GE-Proton7-7/bin/wine --wine-prefix /home/redromnon/Games/Heroic/Prefixes/Kingdom-New-Lands 
[cli] INFO: Logging in...
[Core] INFO: Logging in...
[cli] INFO: Checking for updates...
[Core] INFO: Getting authentication token...
[cli] INFO: Launching cacao...
wineserver: using server-side synchronization.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
Using Bash Launcher 2.5.0
```

**Comment by carn1x** [comment]
> I guess your issue may be solved in the next release.

Great, looking forward to it :D

**Comment by redromnon** [comment]
Has this issue been solved? If not, use the latest v2.7.3.

**Comment by redromnon** [comment]
Closing this issue due to inactivity.
